src/python/grpcio/grpc/_cython/_cygrpc/aio/iomgr/socket.pyx.pxi

```diff
@@ -132,3 +141,38 @@ cdef class _AsyncioSocket:
     cdef void close(self):
         if self.is_connected():
             self._writer.close()
+
+    def _new_connection_callback(self, object reader, object writer):
+        client_socket = _AsyncioSocket.create(self._grpc_client_socket)
```

Comment:
What about having the `reader` and the `writer` as keywords of the `create` function, so having something like this

```python
client_socket = _AsyncioSocket.create(self._grpc_client_socket, reader=reader, writer=writer)
```

At the same time, the `peername` logic could be introduced within the constructor

```python
if writer:
    self._writer = writer
    self._peername = writer.get_extra_info('peername')
else:
    self._writer = None
    self._peername = None
```

At the same time the in the `connect` method I would skip any connection if the `_reader` is already present ( we could use the `is_connected()` method)